src/Core/Framework/Api/EventListener/ErrorResponseFactory.php

```diff
@@ -19,7 +19,7 @@ public function getResponseFromException(\Throwable $exception, bool $debug = fa
         );
 
         $response->setEncodingOptions($response->getEncodingOptions() | \JSON_INVALID_UTF8_SUBSTITUTE);
-        $response->setData(['errors' => $this->getErrorsFromException($exception, $debug)]);
+        $response->setData(['errors' => $this->getErrorsFromException($exception, $debug), 'stack' => $debug ? $exception->getTrace() : null]);
```

Comment:
Can you add a test?

Reply:
can you give me a little hint on where the best place for this test would be?

Reply:
Create a new file at

tests/unit/php/Core/Framework/Api/EventListener/ErrorResponseFactoryTest.php

```php
<?php

namespace Shopware\Tests\Unit\Core\Framework\Api\EventListener;

/**
 * @internal
 *
 * @covers \Shopware\Core\Framework\Api\EventListener\ErrorResponseFactory
 */
class ErrorResponseFactoryTest extends \PHPUnit\Framework\TestCase
{
```

and then write a unit test for this.

You can run it with ./vendor/bin/phpunit tests/unit/php/Core/Framework/Api/EventListener/ErrorResponseFactoryTest.php

Reply:
added test

Reply:
@shyim anything missing here?

Reply:
Can you add an quick changelog?

Reply:
@shyim done